### PR TITLE
Stop building Boot images

### PR DESCRIPTION
**Pull request by alexander-yakushev**
As discussed in #224.

### CODE REVIEW

**Comment by cap10morgan** [comment]
@alexander-yakushev Thanks! Can you split up the `src/` changes from the `target/` changes into separate commits?

**Comment by alexander-yakushev** [comment]
Done!

File: src/docker_clojure/dockerfile.clj [hunk 1]
```diff
@@ -2,8 +2,7 @@
   (:require
    [clojure.java.shell :refer [sh]]
    [clojure.string :as str]
-   [docker-clojure.dockerfile.boot :as boot]
-   [docker-clojure.dockerfile.lein :as lein]
+    [docker-clojure.dockerfile.lein :as lein]
```
Comment (cap10morgan) [inline src/docker_clojure/dockerfile.clj:5]: Fix indentation please

File: README.md [hunk 2]
```diff
@@ -18,13 +18,8 @@ installed. If you want to use tools-deps, specify either `clojure:tools-deps`,
...
 ### boot
 
-Prior to JDK 20 we provided [boot](https://boot-clj.github.io) variants as well. 
-Boot hasn't had a release since 2019, and it is breaking in more and more image
-variants. Boot variants are now deprecated and no new boot images will be
-provided for JDK 20+, it will no longer be installed in the `latest` image, and
-no alpine-based images will be provided for any JDK version (due to breakage).
-As long as the image builds don't break, we will continue providing boot images
-for non-alpine distros for JDK 17 and lower until those versions are EOL'd.
+New [boot](https://boot-clj.github.io) images are no longer provided. The
```
Comment (cap10morgan) [inline README.md:21]: Can you add e.g. "As of 5/2024" to this so folks know roughly when the change was made?

**Review by cap10morgan — CHANGES_REQUESTED** [review]
Couple minor things to address, but looking good overall!

**Comment by alexander-yakushev** [comment]
✅

**Review by cap10morgan — APPROVED** [review]
LGTM

**Pull request merged**